packages/supersearch/src/lib/components/SuperSearch.svelte

```diff
@@ -1,12 +1,34 @@
 <script lang="ts">
 	import CodeMirror, { type ChangeCodeMirrorEvent } from '$lib/components/CodeMirror.svelte';
-	import { placeholder } from '@codemirror/view';
+	import { EditorView, placeholder as placeholderExtension } from '@codemirror/view';
+	import { Compartment } from '@codemirror/state';
+
+	interface Props {
+		placeholder?: string;
+	}
+
+	let { placeholder = '' }: Props = $props();
 
 	let value = $state('');
+	let editorView: EditorView | undefined = $state(undefined);
 
+	let placeholderCompartment = new Compartment();
+	let prevPlaceholder = placeholder;
+
+	const extensions = [placeholderCompartment.of(placeholderExtension(placeholder))];
+	
 	function handleChangeCodeMirror(event: ChangeCodeMirrorEvent) {
 		value = event.value;
 	}
+
+	$effect(() => {
+		if (placeholder !== prevPlaceholder) {
+			editorView?.dispatch({
+				effects: placeholderCompartment.reconfigure(placeholderExtension(placeholder))
+			});
+			prevPlaceholder = placeholder;
```

Comment:
It's a little bit cumbersome having to add an equality check on `placeholder` to prevent the effect from firing on mount – but I cannot think of a better way right now... suggestions are appreciated! :)